### ISSUE TITLE
Some Nomad Snapshot Agent issues

### ISSUE THREAD

**Issue by rberlind** [hashicorp/nomad#8427]
### Nomad version
Nomad 0.12

### Issue

The new Nomad Snapshot Agent has HCL keys taken from the Consul Snapshot Agent that do not match the corresponding Nomad CLI arguments since these are different from the Consul CLI arguments.  It would be better if the actual HCL keys for the Nomad Snapshot Agent matched the Nomad CLI arguments.  Internal consistency within Nomad is more important in my view than consistency between Nomad and Consul, especially since the document, https://www.nomadproject.io/docs/commands/operator/snapshot-agent, only documents the Nomad CLI arguments and expects the reader to infer the meaning of the HCL keys by finding the matching CLI arguments (with the only difference being use of underscores instead of hypens).  Without internal consistency between the Nomad Snapshot Agent HCL keys and the Nomad CLI arguments, the reader must do more guesswork to figure out which HCL keys to use.

Here are some examples of mismatches:

|HCL Key          |CLI Argument|
|--------------|--------------|
|http_addr        |-address        |
|ca_file             |-ca-cert         |
|cert_file          |-client-cert    |
|key_file           |-client-key     |

Additionally, the doc, https://www.nomadproject.io/docs/commands/operator/snapshot-agent, does not document equivalent CLI arguments for the Google storage option. Hopefully, the Nomad agent CLI has actually added an argument like `-google-bucket`.

It also does not show Nomad namespace in the HCL keys. Is it possible to specify the Nomad namespace in HCL? I would think this is required.

Additionally, the use of Consul by the Nomad Snapshot Agent is not made clear.  If Consul is required, then this should be clearly stated. While there is a consul stanza in the HCL, there are no corresponding CLI arguments. In so far as those CLI arguments do not exist, then a section documenting the consul HCL keys should be added. Additionally, the statement "In daemon mode, the agent will also register itself with Nomad as a service" should be "In daemon mode, the agent will also register itself with Consul as a service".  I presume that the write of the doc changed "Consul" to "Nomad" globally, but should not have done so in this case.

**Comment by notnoop** [comment]
Thanks @rberlind for the feedback!  Very good valid points.  I'll follow up.

**Comment by rberlind** [comment]
Thanks @notnoop

**Comment by notnoop** [comment]
Thank you so much for reporting the issues.  I have updated the docs to clarify dependency on Consul, added `-google-bucket` flag, and changed the hcl key address flag to `address`.  The doc change is in https://github.com/hashicorp/nomad/pull/8501/files .

I kept the rest of the inconsistency for tls keys, as it matches the agent config vs CLI flags already; [Nomad agent `tls` configuration](https://www.nomadproject.io/docs/configuration/tls#tls-parameters) use `ca_file`/`cert_file`, while the CLI uses `-ca-cert` in [CLI general options](https://www.nomadproject.io/docs/commands/job/run#general-options).  The current makes it easier to copy hcl agent config to form this or reuse the CLI flags.

To avoid confusion, we can be more explicit in the doc about which flag matches which hcl key.

> It also does not show Nomad namespace in the HCL keys. Is it possible to specify the Nomad namespace in HCL? I would think this is required.

Nope.  The agent will snapshot the entire state, in all namespaces.
